modules/javafx.controls/src/main/java/javafx/scene/control/SkinBase.java

```diff
@@ -238,7 +241,79 @@ protected final void registerChangeListener(ObservableValue<?> property, Consume
         return lambdaChangeListenerHandler.unregisterChangeListeners(property);
     }
 
+    /**
+     * Registers an operation to perform when the given {@code observable} sends an invalidation event.
+     * Does nothing if either {@code observable} or {@code operation} are {@code null}.
+     * If multiple operations are registered on the same observable, they will be performed in the
+     * order in which they were registered.
+     *
+     * @param observable the observable to observe for invalidation events, may be {@code null}
+     * @param operation the operation to perform when the observable sends an invalidation event,
+     *  may be {@code null}
+     * @since 17
+     */
+    protected final void registerInvalidationListener(Observable observable, Consumer<Observable> operation) {
+        if (lambdaChangeListenerHandler == null) {
+            lambdaChangeListenerHandler = new LambdaMultiplePropertyChangeListenerHandler();
+        }
+        lambdaChangeListenerHandler.registerInvalidationListener(observable, operation);
+    }
+
+    /**
+     * Unregisters all operations that have been registered using
+     * {@link #registerInvalidationListener(Observable, Consumer)}
+     * for the given {@code observable}. Does nothing if {@code observable} is {@code null}
+     *
+     * @param observable the observable for which the registered operations should be removed,
+     *  may be {@code null}
+     * @return a composed consumer that performs all removed operations, or
+     *  {@code null} if none have been registered or the observable is {@code null}
```

Comment:
Somehow my comment seems to have been deleted, so here it is again, slightly modified, along with a suggestion.

I haven't gone back and done a detailed review yet, but I like the overall changes. The one thing I did notice is that the language used to describe the return value of `unregisterInvalidationListeners` and `unregisterListChangeListeners` is different:

```
unregisterInvalidationListeners:
     * @return a composed consumer that performs all removed operations, or
     *  {@code null} if none have been registered or the observable is {@code null}
```

```
unregisterListChangeListeners:
     * @return A single chained {@link Consumer} consisting of all {@link Consumer consumers} registered through
     *      {@link #registerListChangeListener(ObservableList, Consumer)}. If no consumers have been registered on this
     *      list, null will be returned.
```

I find it confusing to say that the returned list "performs all removed operations". Some might interpret this to mean that is is somehow necessary to call the returned chained consumer as part of the removal, which isn't what you meant to say.

How about something like this for both newly added unregister methods?

```
     * @return a composed consumer consisting of all previously registered consumers, or
     * {@code null} if none have been registered or the observable is {@code null}
```

Reply:
@Kevin 

don't worry, I have seen your comment (and been thinking about it :) - but also can't find it right now (nor can I find Nir's comment which started the overhaul ..), no idea what happened.

Seeing your suggestion, it's similar to what keeps running in my mind.

Reply:
Maybe GitHub decided to prune it. 😃 

Btw, you tagged the wrong "Kevin" above.

Reply:
ahh .. looks like you marked the conversation that contained the original comment as resolved: https://github.com/openjdk/jfx/pull/409#discussion_r604387696 ..

Reply:
I guess I didn't look closely enough. Mystery solved, though.

Reply:
> I haven't gone back and done a detailed review yet, but I like the overall changes. The one thing I did notice is that the language used to describe the return value of `unregisterInvalidationListeners` and `unregisterListChangeListeners` is different:
> 

just a side-note: technically, both are different because the neither the new listChangeListener nor the old changeListener methods are updated yet (we are still at the first bullet of the plan, lazy me will update all to the same as the invalidationListener wording, once we agree on it :)

So what we are arguing right now is the difference between new (= invalidationListener) and old (= listChangeListener == c&p from changeListener) method doc.

> ```
> unregisterInvalidationListeners:
>      * @return a composed consumer that performs all removed operations, or
>      *  {@code null} if none have been registered or the observable is {@code null}
> ```
> 
> ```
> unregisterListChangeListeners:
>      * @return A single chained {@link Consumer} consisting of all {@link Consumer consumers} registered through
>      *      {@link #registerListChangeListener(ObservableList, Consumer)}. If no consumers have been registered on this
>      *      list, null will be returned.
> ```
> 
> I find it confusing to say that the returned list "performs all removed operations". Some might interpret this to mean that is is somehow necessary to call the returned chained consumer as part of the removal, which isn't what you meant to say.
> 
> How about something like this for both newly added unregister methods?
> 
> ```
>      * @return a composed consumer consisting of all previously registered consumers, or
>      * {@code null} if none have been registered or the observable is {@code null}
> ```

agree: the _performs_ might be misleading. And I like your suggestion, though I modified it a bit to be consistent with the rest of the changed doc which focuses on "operations" (a term taken from Consumer api): 

     * @return a composed consumer representing all previously registered operations, or
     *  {@code null} if none have been registered or the observable is {@code null}

Reply:
I like this version, using "registered operations".